packages/x-tree-view/src/internals/plugins/useTreeViewSelection/useTreeViewSelection.types.ts

```diff
@@ -3,52 +3,56 @@ import type { DefaultizedProps, TreeViewPluginSignature } from '../../models';
 import { UseTreeViewItemsSignature } from '../useTreeViewItems';
 import { UseTreeViewExpansionSignature } from '../useTreeViewExpansion';
 
-export interface UseTreeViewSelectionInstance {
+export interface UseTreeViewSelectionPublicAPI {
+  /**
+   * Select or deselect an item.
+   * @param {object} params The params of the method.
+   * @param {React.SyntheticEvent} params.event The DOM event that triggered the change.
+   * @param {string} params.itemId The id of the item to select or deselect.
+   * @param {boolean} params.keepExistingSelection If `true`, the other already selected items will remain selected, otherwise, they will be deselected. This parameter is only relevant when `multiSelect` is `true`
+   * @param {boolean | undefined} params.isSelected If `true` the item will be selected. If `false` the item will be deselected. If not defined, the item's new selection status will be the opposite of its current one.
```

Comment:
The naming of `isSelected` is slightly confusing to me. I would expect something that begins with `is[...]` to define a state, but here isSelected defines what should **happen** to the item (if it should be selected or not). Is there a drawback to renaming this to something like `selectItem`?

Reply:
`shouldBeSelected`? To follow a boolean naming :angel:

Reply:
sounds good to me 👌